### PR TITLE
Removing SetEnvKeyTransformer in favor of typed helpers

### PR DESCRIPTION
**Pull request by hush-hush**
### What does this PR do?

This commit remove the last usage of SetEnvKeyTransformer.

This includes moving the dogstatsd logic from the configuration into dogstatsd itself.

### QA steps

For AML: test that mapping settings still works as before for dogstatsd.

### CODE REVIEW

**Review by brycekahle — APPROVED** [review]

**Review by robertjli — APPROVED** [review]

**Review by gh123man — APPROVED** [review]

File: comp/dogstatsd/mapper/mapper.go [hunk 1]
```diff
@@ -25,6 +23,29 @@ const (
 	matchTypeRegex    = "regex"
 )
 
+//
+// Those two struct are used to pull data from the configuration into typed struct. We currently load the data from the
+// configurarion into MappingProfileConfig and then convert it to MappingProfile.
```
Comment (dustmop) [inline comp/dogstatsd/mapper/mapper.go:28]:
```suggestion
// configuration into MappingProfileConfig and then convert it to MappingProfile.
```

File: comp/dogstatsd/mapper/mapper.go [hunk 2]
```diff
@@ -25,6 +23,29 @@ const (
...
+//
+// Both MappingProfileConfig and MetricMappingConfig will be removed once we remove `UnmarshalKey` method from the
```
Comment (dustmop) [inline comp/dogstatsd/mapper/mapper.go:30]: How about updating this to

```suggestion
// Both MappingProfileConfig and MetricMappingConfig will no longer use `mapstructure` once we remove `UnmarshalKey` method from the config interface
```

File: comp/dogstatsd/server/server.go [hunk 3]
```diff
@@ -29,7 +29,7 @@ import (
 	replay "github.com/DataDog/datadog-agent/comp/dogstatsd/replay/def"
 	serverdebug "github.com/DataDog/datadog-agent/comp/dogstatsd/serverDebug"
 	"github.com/DataDog/datadog-agent/pkg/aggregator"
-	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/config/model"
```
Comment (dustmop) [inline comp/dogstatsd/server/server.go:32]: Why change this to model?

File: comp/dogstatsd/server/server.go [hunk 4]
```diff
@@ -452,9 +452,9 @@ func (s *server) start(context.Context) error {
 
 	cacheSize := s.config.GetInt("dogstatsd_mapper_cache_size")
 
-	mappings, err := config.GetDogstatsdMappingProfiles()
+	mappings, err := getDogstatsdMappingProfiles(s.config)
 	if err != nil {
-		s.log.Warnf("Could not parse mapping profiles: %v", err)
+		s.log.Warnf(err.Error())
```
Comment (dustmop) [inline comp/dogstatsd/server/server.go:457]:
```suggestion
		s.log.Warn(err)
```

**Review by dustmop — APPROVED** [review]
LGTM, with a few suggestions

File: pkg/config/aliases.go [hunk 5]
```diff
@@ -222,11 +220,6 @@ func GetBindHost() string {
 	return pkgconfigsetup.GetBindHost(Datadog())
 }
 
-// GetDogstatsdMappingProfiles Alias using Datadog config
-func GetDogstatsdMappingProfiles() ([]MappingProfile, error) {
-	return pkgconfigsetup.GetDogstatsdMappingProfiles(Datadog())
-}
```
Comment (GustavoCaso) [inline pkg/config/aliases.go:228]: 👏

File: pkg/config/setup/process.go [hunk 6]
```diff
@@ -227,9 +224,11 @@ func loadProcessTransforms(config pkgconfigmodel.Config) {
 		} else if enabled, _ := strconv.ParseBool(procConfigEnabled); enabled { // "true"
 			config.Set("process_config.process_collection.enabled", true, pkgconfigmodel.SourceAgentRuntime)
 			config.Set("process_config.container_collection.enabled", false, pkgconfigmodel.SourceAgentRuntime)
+			config.Set("process_config.enabled", "true", pkgconfigmodel.SourceAgentRuntime)
 		} else { // "false"
 			config.Set("process_config.process_collection.enabled", false, pkgconfigmodel.SourceAgentRuntime)
 			config.Set("process_config.container_collection.enabled", true, pkgconfigmodel.SourceAgentRuntime)
+			config.Set("process_config.enabled", "disabled", pkgconfigmodel.SourceAgentRuntime)
```
Comment (GustavoCaso) [inline pkg/config/setup/process.go:231]: Q: Do we use `disbale` string is a standard for the agent to express a feature is disable? Do we support other values?

**Review by GustavoCaso — APPROVED** [review]

File: comp/dogstatsd/server/server.go (hunk 3 above)
Comment (hush-hush) [reply comp/dogstatsd/server/server.go:32]: So we stop using the aliases but the real package in the config space. This reduce dependencies and highligh the real dependency graph.

File: pkg/config/setup/process.go (hunk 6 above)
Comment (hush-hush) [reply pkg/config/setup/process.go:231]: This is purely keeping the current behavior. We don't have any guidance for this king of behavior and each setting was handled independently by each team.
I kept the behavior the same to be backward compatible.

**Pull request merged**